dataset.py

```diff
@@ -85,16 +108,47 @@ def add(self, dataset: str, timestamp: float, modification: Modification):
             timestamp: The timestamp of the modification.
             modification: Modification dict, e.g., {"action": "append", "x": value}.
         """
-        self._modifications[dataset].push(timestamp, modification)
+        queue = self._modifications.get(dataset, None)
+        if queue is None:
+            logger.error("Cannot add modification to dataset %s since it does not exist.", dataset)
+            return
+        queue.push(timestamp, modification)
 
-    def since(self, dataset: str, timestamp: float) -> Tuple[float, Tuple[Modification]]:
+    def since(self, dataset: str, timestamp: float) -> Tuple[float, Tuple[Modification, ...]]:
         """Returns the latest timestamp and modifications since the given timestamp.
 
         See SortedQueue.tail() for details.
         
         Args:
             dataset: Target dataset name.
+              If dataset does not exist, it returns (-1, ()).
             timestamp: The last timestamp of the previous update.
               Any modifications added after this timestamp will be returned.
         """
-        return self._modifications[dataset].tail(timestamp)
+        queue = self._modifications.get(dataset, None)
+        if queue is None:
+            logger.error("Cannot call since() for dataset %s since it does not exist.", dataset)
+            return (-1, ())
+        return queue.tail(timestamp)
+
+
+def notify_callback(tracker: DatasetTracker, mod: Dict[str, Any]):
+    """Adds modification to the tracker called as notify_cb() of sipyco system.
+    
+    Args:
+        tracker: Target dataset tracket object.
+        mod: The argument of notify_cb() called by sipyco.sync_struct.Subscriber.
+    """
+    action = mod["action"]
+    if action == "init":
+        return
+    if not mod["path"]:
+        if action == "setitem":
+            tracker.add_dataset(mod["key"])
```

Comment:
Don't we need to set dataset contents?
In according to [sync_struct.py](https://github.com/m-labs/sipyco/blob/c4f18adb658e3a546319425750cebeb8c88a016a/sipyco/sync_struct.py#L79), it adds a key, and also sets the value in the corresponding key.

Reply:
Since `DatasetTracker` only tracks modifications, I think it does not have to set the contents.
It only contains the modifications, and the user should call `get_master_dataset()` to get the dataset itself. Then the user will fetch modifications to update their dataset.
Do you understand what I mean?

Reply:
I see! Then, there is no problem :)